qt/aqt/utils.py

```diff
@@ -459,17 +459,71 @@ def ensureWidgetInScreenBoundaries(widget):
         widget.move(x, y)
 
 
-def saveState(widget, key):
+def saveState(widget, key: str):
     key += "State"
     aqt.mw.pm.profile[key] = widget.saveState()
 
 
-def restoreState(widget, key):
+def restoreState(widget, key: str):
     key += "State"
     if aqt.mw.pm.profile.get(key):
         widget.restoreState(aqt.mw.pm.profile[key])
 
 
+def saveIsChecked(widget, key: str):
+    key += "IsChecked"
+    aqt.mw.pm.profile[key] = widget.isChecked()
+
+
+def restoreIsChecked(widget, key: str):
+    key += "IsChecked"
+    if aqt.mw.pm.profile.get(key) is not None:
+        widget.setChecked(aqt.mw.pm.profile[key])
+
+
+def saveComboIndex(widget: QComboBox, key: str):
+    textKey = key + "ComboActiveText"
+    indexKey = key + "ComboActiveIndex"
+    aqt.mw.pm.session[textKey] = widget.currentText()
+    aqt.mw.pm.session[indexKey] = widget.currentIndex()
+
+
+def restoreComboIndex(widget: QComboBox, history: List[str], key: str):
+    textKey = key + "ComboActiveText"
+    indexKey = key + "ComboActiveIndex"
+    text = aqt.mw.pm.session.get(textKey)
+    index = aqt.mw.pm.session.get(indexKey)
+    if text is not None and index is not None:
+        if index < len(history) and history[index] == text:
+            widget.setCurrentIndex(index)
+
+
+def saveComboHistory(comboBox: QComboBox, history: List[str], name: str):
+    name += "BoxHistory"
```

Comment:
- Since we now have a mix of routines that save permanently and save for the session, perhaps it would be better to encode that in the naming, eg "save_combo_index_for_session"
- Since these free-functions are just modifying state in the pm object, perhaps it makes more sense for them to be placed inside that class, or in a separate SessionState class accessible from mw that gets recreated on profile load.

Reply:
Only `save_combo_index` saves on the `session` object, `save_combo_history` uses the `session` object as a hint to not fill the results when the user has restarted Anki. Unless we have a specific problem to solve (requiring [separation of concerns](https://en.wikipedia.org/wiki/Separation_of_concerns)), a dictionary attribute on ProfileManager should be more than enough.

We could move these functions into the `Profile` object as class members, but only if we also move the existing functions as `saveState` and `saveSplitter` with it because they do the same things so it will be very nice for new developers to find all of them in a single place.

Reply:
Please at least add for_session for the one routine that doesn't use the profile, as it's confusing otherwise.